src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.circulation.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.domain.representations.CheckOutByBarcodeRequest.ITEM_BARCODE;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.library.ClosedLibraryStrategyService;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Checkout strategy for the loanable items.
+ * Calculates due date based on loan policy and closed library due date management.
+ * Refuses request if an item is not loanable
+ */
+public class RegularCheckOutStrategy implements CheckOutStrategy {
```

Comment:
I'm not keen on `regular` however I am unable to think of an alternative at this point. I do think this is a hint that there is a missing abstraction here. I still think this is a good step forwards.

Reply:
Feel the same, but hasn't come up with better naming/abstraction in the scope of this pull request.